### PR TITLE
added resting to chateau mantegna

### PR DESCRIPTION
**Pull request by ktpolanski**

### CODE REVIEW

**Review by horrible-little-slime — DISMISSED** [review]

File: src/resources/2015/ChateauMantegna.ts [hunk 1]
```diff
@@ -18,6 +26,17 @@ export function fightPainting(): string {
   return runCombat();
 }
 
+export function rest(): void {
+  visitUrl("place.php?whichplace=chateau&action=chateau_restbox");
+}
+
+export function restIfFree(): boolean {
```
Comment (gausie) [inline src/resources/2015/ChateauMantegna.ts:33]: This function should probably be taken out into a general library function, and could even take a callback that would dictate what rest it did

**Comment by ktpolanski** [comment]
Once this suggestion was made, I talked the issue over with phred. I'm not a huge fan of a generic `restIfFree()` with a callback as it's not the user friendliest in my opinion. However, it does make the code future-proof, and seems like it can be easily locally used for specific rest types as per phred's recent idea.

**Comment by gausie** [comment]
@ktpolanski sorry for letting this get stale - want to fix the conflicts?

**Comment by gausie** [comment]
Going to consider this dead. Please please feel free to reopen and we'll review quickly

**Pull request closed without merging**